test/test_allgather.jl

```diff
@@ -10,7 +10,7 @@ end
 
 comm = MPI.COMM_WORLD
 
-for typ in (isdefined(:UnionAll) ? Base.uniontypes(MPI.MPIDatatype) : MPI.MPIDatatype.types)
```

Comment:
Should just be `Base.uniontypes(MPI.MPIDatatype)`, since we've dropped pre 0.6 support.